src/Office-Add-in-ASPNET-SSO-WebAPI/Scripts/HomeES6.js

```diff
@@ -0,0 +1,209 @@
+            break;
+        case 13002:
+            // The user aborted the consent prompt. If the add-in cannot be effectively used when consent
+            // has not been granted, then the first call of getAccessToken should pass the `allowConsentPrompt: true` option.
+            showResult(["You can use many of the add-ins functions even though you have not granted consent. If you want to grant consent, press the Get OneDrive File Names button again."]);
+            break;
+        case 13006:
+            // Only seen in Office on the Web.
+            showResult(["Office on the Web is experiencing a problem. Please sign out of Office, close the browser, and then start again."]);
+            break;
+        case 13008:
+            // Only seen in Office on the Web.
+            showResult(["Office is still working on the last operation. When it completes, try this operation again."]);
+            break;
+        case 13010:
+            // Only seen in Office on the Web.
+            showResult(["Follow the instructions to change your browser's zone configuration."]);
+            break;
+        default:
+            // For all other errors, including 13000, 13003, 13005, 13007, 13012, and 50001, fall back
+            // to non-SSO sign-in.
+            dialogFallback();
+            break;
+    }
+}
+
+function handleServerSideErrors(result) {
+
+	// Our special handling on the server will cause the result that is returned
+	// from a AADSTS50076 (a 2FA challenge) to have a Message property but no ExceptionMessage.
+	var message = JSON.parse(result.responseText).Message;
+
+
+	// Results from other errors (other than AADSTS50076) will have an ExceptionMessage property.
+	var exceptionMessage = JSON.parse(result.responseText).ExceptionMessage;
+
+	// Microsoft Graph requires an additional form of authentication. Have the Office host 
+	// get a new token using the Claims string, which tells AAD to prompt the user for all 
+	// required forms of authentication.
+	if (message) {
+		if (message.indexOf("AADSTS50076") !== -1) {
+			var claims = JSON.parse(message).Claims;
+			var claimsAsString = JSON.stringify(claims);
+			getDataWithToken({ authChallenge: claimsAsString });
+			return;
+		}
+	}
+
+	if (exceptionMessage) {
+
+        // On rare occasions the bootstrap token is unexpired when Office validates it,
+        // but expires by the time it is sent to AAD for exchange. AAD will respond
+        // with "The provided value for the 'assertion' is not valid. The assertion has expired."
+        // Retry the call of getAccessToken (no more than once). This time Office will return a 
+        // new unexpired bootstrap token.
+        if ((exceptionMessage.indexOf("The provided value for the 'assertion' is not valid. The assertion has expired.") !== -1)
```

Comment:
Do we need to check the exact string? This will certainly break in the future.

Reply:
Switched to the number AADSTS500133